### ISSUE TITLE
Not opening file manager on Elementary OS

### ISSUE THREAD

**Issue by ghost** [shiftkey/desktop#293]
### Describe the bug

Click **Show in your File Manager** on Elementary OS opens in **Code**(io.elementary.code)
This appears to happen when using Elementary's default coding app as default, this wasn't an issue before when I was using Visual Studio Code.

### Version & OS

Version: 2.5.2-linux1
OS: elementary OS 5.1.5 Hera

### Steps to reproduce the behavior

1. Open Github Desktop
2. Select a repo
3. Click **Show in File Manager**

### Expected behavior

When clicking **Show in your File Manager**, the file manager opens in the redirecting to the proper folder that the repo is in.

### Actual behavior

When clicking **Show in your File Manager**, the code editor(io.elementary.code) opens instead of the file manager. This also includes clicking **Show Logs**.

### Additional context

Appears Github desktop has an Intergration setting. Would be nice to add the file manager option there and also setting a custom one. So I would put in  `io.elementary.files` as an example.

**Comment by shiftkey** [comment]
@KorbsStudio thanks for the report. I don't have Elementary OS handy to test this myself so I'm going by the details provided.

This is a tricky bug because the API we're calling - [`shell.openExternal`](https://www.electronjs.org/docs/api/shell#shellopenexternalurl-options) - is part of Electron itself. 

https://github.com/electron/electron/blob/c6c022dc46258b439c311da7feaa15d24dc08b16/shell/browser/electron_browser_client.cc#L1139-L1144

```cpp
void OnOpenExternal(const GURL& escaped_url, bool allowed) {
  if (allowed) {
    platform_util::OpenExternal(
        escaped_url, platform_util::OpenExternalOptions(), base::DoNothing());
  }
}
```

And Electron is just calling into Chromium itself, and it looks like this wraps `xdg-open`:

https://source.chromium.org/chromium/chromium/src/+/master:chrome/browser/platform_util_linux.cc;l=184-190?q=OpenExternal&ss=chromium%2Fchromium%2Fsrc&originalUrl=https:%2F%2Fcs.chromium.org%2F

```cpp
void OpenExternal(Profile* profile, const GURL& url) {
  DCHECK_CURRENTLY_ON(BrowserThread::UI);
  if (url.SchemeIs("mailto"))
    XDGEmail(url.spec());
  else
    XDGOpen(base::FilePath(), url.spec());
}
```
I'm hazy on what to poke at about your xdg settings here to understand whether something is misconfigured, and it's gonna take a few days for me to find the bandwidth to dig into that area again.

**Comment by shiftkey** [comment]
Wait, I misread a part of the above steps. Here's the method that we use which wraps the Electron APIs:

https://github.com/shiftkey/desktop/blob/48d13371876db41480cd168862b69aa17052162d/app/src/main-process/shell.ts#L4-L27

Notice the problem? For non-Darwin OSes, we actually call [`shell.openItem`](https://github.com/electron/electron/blob/v7.1.8/docs/api/shell.md#shellopenitemfullpath) here - that might explain the difference in behaviour you're seeing.

**Comment by ghost** [comment]
Alright thanks. So this can be fixed?

**Comment by shiftkey** [comment]
@KorbsStudio apologies for the delay here. I had a look at fixing this and it seems things have changed upstream, and [`shell.openPath`](https://www.electronjs.org/docs/api/shell#shellopenpathpath) is now used. The original `openItem` method was deprecated in Electron v9 so I'd be curious if this has been fixed for you with the `2.6.2-linux1` release.

**Comment by ghost** [comment]
Seems to be working now.
